src/test-kit/OptionButtons/sortByButton-driver.ts

```diff
@@ -0,0 +1,40 @@
+import { Locator, Page } from 'playwright-chromium';
+import { selectors } from '../../test-kit/OptionButtons/selectors';
+
+export class SortByButtonDriver {
+   private sortByButton: Locator;
+   private dropDownOptionList: Locator;
```

Comment:
Two options:
1. make a function that will return the locator instead of setting the as privates (which I prefer).
2. implement the privates in the constructor the same way you did with the `page`.

Reply:
I am not sure I understand the second option. The first one is clear.

Reply:
```ts
 constructor(private page: Page, private sortByButton: Locator, private dropDownOptionList: Locator) {
...
      );
   }
```

Reply:
This one preserves the same logic that you did here, I disagree with the logic that assumes that those locators are mostly used in this driver so therefore let's save them as private.
but I'm not totally against it so feel free to do it.

Reply:
Sorry, but I still don’t completely understand... `sortByButton` and `dropDownOptionList` are constructed inside the constructor using the page parameter. Are you suggesting passing them as arguments like this?
```markdown
sortByButtonDriver = new SortByButtonDriver(
  page, 
  page.locator(selectors.sortByButton),
  page.locator(selectors.sortbyButtonDropDownList)
);
```